src/shell-interface.h

```diff
@@ -87,17 +87,21 @@ struct ShellExternalInterface : ModuleInstance::ExternalInterface {
   } memory;
 
   std::unordered_map<Name, std::vector<Literal>> tables;
+  std::map<Name, ModuleInstance*> linkedInstances;
 
   ShellExternalInterface() : memory() {}
   virtual ~ShellExternalInterface() = default;
 
-  void init(Module& wasm, ModuleInstance& instance) override {
+  void init(Module& wasm,
+            ModuleInstance& instance,
+            std::map<Name, ModuleInstance*> linkedInstances) override {
```

Comment:
```suggestion
            std::map<Name, ModuleInstance*> linkedInstances = {}) override {
```
if that compiles it could save the work of typing `{}` in most callers. but not sure if it will...